FirebaseDynamicLinks/Sources/FIRDynamicLinkNetworking.m

```diff
@@ -101,6 +102,45 @@ - (instancetype)initWithAPIKey:(NSString *)APIKey
   return self;
 }
 
++ (nullable NSError *)extractErrorForShortLink:(NSURL *)url
+                                          data:(NSData *)data
+                                      response:(NSURLResponse *)response
+                                         error:(nullable NSError *)error {
+  if (error) {
+    return error;
+  }
+
+  NSInteger statusCode = [(NSHTTPURLResponse *)response statusCode];
+  NSError *customError = nil;
+
+  if (![response isKindOfClass:[NSHTTPURLResponse class]]) {
+    customError =
+        [NSError errorWithDomain:kGenericErrorDomain
+                            code:0
+                        userInfo:@{@"message" : @"Response should be of type NSHTTPURLResponse."}];
+  } else if (statusCode < 200 || statusCode >= 300) {
+    NSDictionary *result = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
```

Comment:
Don't we need to check data before using?

Reply:
Right. Added the check here: https://github.com/firebase/firebase-ios-sdk/pull/5853/commits/73f3bb291fbd0b37c7e99fe95033c58e0c03fbd2